### ISSUE TITLE
Additional ClientTimeout parameter support on async API calls - sock_read, sock_connect

### ISSUE THREAD

**Issue by suhjohn** [openai/openai-python#336]
### Describe the feature or improvement you're requesting

Want a logic to retry API if sock_read on stream is not working in <2 seconds. Sometimes this happens on a web-server due to various issues with the requesting library (requests.request / aiohttp). Currently only
```
            timeout = aiohttp.ClientTimeout(
                connect=request_timeout[0],
                total=request_timeout[1],
            )
```

is supported. Perhaps accepting additional lengths on request_timeout, or adding a new kwarg for timeout object, or accepting a dictionary instead a tuple on request_timeout would be helpful.

**Comment by ShantanuNair** [comment]
@suhjohn I'm exploring this issue currently. Is this PR relevant here? 
https://github.com/openai/openai-python/pull/387

**Comment by ShantanuNair** [comment]
@suhjohn Apparently fixed in the v1.0.0 beta: https://github.com/openai/openai-python/discussions/631#discussioncomment-7206935
